src/main/java/seedu/address/model/employee/EmployeeUnderFilteredProjectsPredicate.java

```diff
@@ -0,0 +1,47 @@
+package seedu.address.model.employee;
+
+import java.util.function.Predicate;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Employee} is under any of the specified filtered projects.
+ */
+public class EmployeeUnderFilteredProjectsPredicate implements Predicate<Employee> {
+
+    private final ObservableList<Project> filteredProjects;
+
+    public EmployeeUnderFilteredProjectsPredicate(ObservableList<Project> filteredProjects) {
+        this.filteredProjects = filteredProjects;
+    }
+
+    @Override
+    public boolean test(Employee employee) {
+        return filteredProjects.stream()
+                .anyMatch(project -> project.getEmployees().stream()
+                        .anyMatch(otherEmployee -> otherEmployee.isSameEmployee(employee)));
```

Comment:
Personally, I feel that with the class documentation as well as the descriptive variable names, it should not be too difficult to understand the above at first glance. Also, streams could help to make the function code more concise.

Reply:
To amend what I just said, since we are now using `UniqueEmployeeList` in a project, I can make the above function shorter using the `contains` method in `UniqueEmployeeList`.